marketplace/src/benchmarking.rs

```diff
@@ -17,236 +17,166 @@
-	verify {
-		assert_eq!(T::NFTExt::owner(nft_id), Some(alice));
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), true);
-	}
-
-	unlist_nft {
-		let (mkp_id, _, nft_id) = prepare_benchmarks::<T>();
-
-		let alice = get_origin::<T>("ALICE");
-		let price: BalanceOf<T> = 100u32.into();
-		drop(Marketplace::<T>::list_nft(alice.clone().into(), nft_id, price, Some(mkp_id)));
-
-	}: _(alice.clone(), nft_id)
-	verify {
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), false);
-	}
-
-	buy_nft {
-		let (mkp_id, _, nft_id) = prepare_benchmarks::<T>();
-
-		let bob: T::AccountId = get_account::<T>("BOB");
-		let price: BalanceOf<T> = 0u32.into();
-
-		drop(Marketplace::<T>::list_nft(get_origin::<T>("ALICE").into(), nft_id, price, Some(mkp_id)));
-	}: _(RawOrigin::Signed(bob.clone().into()), nft_id)
-	verify {
-		assert_eq!(T::NFTExt::owner(nft_id), Some(bob));
-		assert_eq!(NFTsForSale::<T>::contains_key(nft_id), false);
-	}
-
 	create_marketplace {
 		prepare_benchmarks::<T>();
-
 		let alice: T::AccountId = get_account::<T>("ALICE");
-		let mkp_id = Marketplace::<T>::marketplace_id_generator() + 1;
-	}: _(RawOrigin::Signed(alice.clone().into()), MarketplaceType::Public, 0, bounded_vec![20, 30, 40], bounded_vec![], bounded_vec![], bounded_vec![])
+		let alice_origin = origin::<T>("ALICE");
+		let marketplace_offchain_data =
+			BoundedVec::try_from(vec![1; T::OffchainDataLimit::get() as usize])
+				.expect("It will never happen.");
+	}: _(alice_origin, MarketplaceType::Public, Some(MarketplaceFee::Percentage(PERCENT_50)), Some(MarketplaceFee::Percentage(PERCENT_50)), Some(marketplace_offchain_data))
 	verify {
-		assert_eq!(Marketplaces::<T>::contains_key(mkp_id), true);
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().owner, alice);
-		assert_eq!(MarketplaceIdGenerator::<T>::get(), mkp_id);
-	}
-
-	add_account_to_allow_list {
-		let (_, mkp_id, _) = prepare_benchmarks::<T>();
-
-		let bob: T::AccountId = get_account::<T>("BOB");
-		let bob_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(bob.clone());
-
-
-	}: _(get_origin::<T>("ALICE"), mkp_id, bob_lookup.into())
-	verify {
-		let allow_list: AccountVec<T> = bounded_vec![bob];
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().allow_list, allow_list);
-	}
-
-	remove_account_from_allow_list {
-		let (_, mkp_id, _) = prepare_benchmarks::<T>();
-
-		let alice = get_origin::<T>("ALICE");
-		let bob: T::AccountId = get_account::<T>("BOB");
-		let bob_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(bob.clone());
-		drop(Marketplace::<T>::add_account_to_allow_list(alice.clone().into(), mkp_id, bob_lookup.clone()));
-
-	}: _(alice.clone(), mkp_id, bob_lookup)
-	verify {
-		let allow_list: AccountVec<T> = bounded_vec![];
-		assert_eq!(Marketplaces::<T>::get(mkp_id).unwrap().allow_list, allow_list);
+		let marketplace_id = Marketplace::<T>::get_next_marketplace_id() - 1;
+		assert_eq!(Marketplaces::<T>::contains_key(marketplace_id), true);
```

Comment:
Probably was done before my me but you don't need this line. It's going to crash anyway on like 75 if an the id does not exist.